force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,50 @@
+trigger OpportunityTrigger on Opportunity(after update, before update) {
+	if (Trigger.isAfter) {
+		List<Task> taskList = new List<Task>();
+		List<Messaging.SingleEmailMessage> mailList = new List<Messaging.SingleEmailMessage>();
+		Map<Id, Opportunity> oldMap = Trigger.oldMap;
+		Map<Id, Partner__c> parMap = new Map<Id, Partner__c>(
+			[SELECT Address__c, Name, Integration__c, id FROM Partner__c]
```

Comment:
What if we have more than 50000 Partners in the org? Then we exceed limit and get exceprion.We should identify which exact partners we're looking for.
For example go in for loop by all opportunities and collect needed Partner ids in the Set<Id> partnerIds (adding from oppNew.Partner__c) and then query only needed partners using WHERE Id IN :partnerIds